associations/has_one_association.go

```diff
@@ -100,3 +100,34 @@ func (h *hasOneAssociation) AfterSetup() error {
 
 	return fmt.Errorf("could not set '%s' to '%s'", ownerID, fval)
 }
+
+func (h *hasOneAssociation) AfterProcess() AssociationStatement {
+	otherIDField := "ID"
+	id := h.ownedModel.FieldByName(otherIDField).Interface()
+
+	ownerIDField := "ID"
+	ownerID := reflect.Indirect(reflect.ValueOf(h.owner)).FieldByName(ownerIDField).Interface()
+
+	ids := []interface{}{ownerID}
+
+	if !IsZeroOfUnderlyingType(id) {
+		ids = append(ids, id)
+	} else {
+		return AssociationStatement{
```

Comment:
Reverse the if condition, so you can drop the else block.

Reply:
I addressed this in the latest commit... not sure why it didn't mark it as `Outdated`...